src/app/map/Chat/_hooks/useAIChat.ts

```
@@ -0,0 +1,131 @@
+interface UseAIChatOptions {
+  temperature?: number
+  maxTokens?: number
+  compositionConfig?: CompositionConfig
+}
+
+export function useAIChat(options: UseAIChatOptions = {}) {
+  const chatState = useChatState()
+  // Try to get cache context - might not be available in tests
+  let cacheState: ReturnType<typeof useMapCacheContext>['state'] | null = null
+  try {
+    const context = useMapCacheContext()
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)